### PR TITLE
`current`-module to handle ContextVars, new `current.user` ContextVar

### PR DESCRIPTION
**Pull request by ArneGudermann**
This PR replaces `utils.current.*` and `utils.getCurrentUser()` with a `current` module to keep "`current`"-values at one place, and not as part of the `utils` module.

Relates to #632 and #633

### CODE REVIEW

**Comment by phorward** [comment]
@ArneGudermann can you please fix the conflicts that came in with the latest pull request merges first? After that I can start a review.

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, I've found one invalid place (`Request` written in upper-case-order) which must be fixed.

Furthermore, I would like to propose to use the walrus-operator more often in situations like
```python
        user = current.user.get()
        if user:
            ...
```
and
```python
        user = current.user.get()
        if not user:
            ...
```
and use the modern syntax, which is
```python
        if user := current.user.get()
            ...
```
and
```python
        if not (user := current.user.get()):
            ...
```

Can you please apply this?

File: core/prototypes/singleton.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def canEdit(self) -> bool:
 
         :returns: True, if editing is allowed, False otherwise.
         """
-        user = utils.getCurrentUser()
+        user = current.user.get()
 
         if not user:
```
[the remark this answers is not in the document]
Comment (phorward) [reply core/prototypes/singleton.py:226]: please replace all occurrences like this in the entire PR (there are plenty of them)

**Review by sveneberth — CHANGES_REQUESTED** [review]
I think the `user` must be set after each login and logout too.

**Comment by phorward** [comment]
> I think the `user` must be set after each login and logout too.

Haha, exactly this was my thought when I reviewed it right now ;-D

**Review by phorward — CHANGES_REQUESTED** [review]
@ArneGudermann please check my suggestions from #652 to integrate them here.

**Comment by phorward** [comment]
@ArneGudermann I've accidentally merged #652 into this branch some minutes ago. I hope you're fine with it and all reviewers comply with the changes. If not, please annotate here.

**Review by phorward — DISMISSED** [review]

**Comment by ArneGudermann** [comment]
@phorward It looks okay for me.

**Comment by phorward** [comment]
> I think the `user` must be set after each login and logout too.

Dear Mr. @sveneberth can you please re-approve this PR?
The context-variable is set in Request and on the user-module appropriately now.

**Review by phorward — APPROVED** [review]

**Review by sveneberth — APPROVED** [review]
awesome!

**Pull request merged**